geoh5py/objects/block_model.py

```diff
@@ -67,210 +90,104 @@ def centroids(self) -> np.ndarray | None:
                 [x_N, y_N, z_N]
             ]
         """
-        if (
-            getattr(self, "_centroids", None) is None
-            and self.u_cells is not None
-            and self.v_cells is not None
-            and self.z_cells is not None
-            and self.origin is not None
-        ):
-            cell_center_u = np.cumsum(self.u_cells) - self.u_cells / 2.0
-            cell_center_v = np.cumsum(self.v_cells) - self.v_cells / 2.0
-            cell_center_z = np.cumsum(self.z_cells) - self.z_cells / 2.0
-
+        if getattr(self, "_centroids", None) is None:
+            cell_center_u = self.local_axis_centers("u")
+            cell_center_v = self.local_axis_centers("v")
+            cell_center_z = self.local_axis_centers("z")
             angle = np.deg2rad(self.rotation)
             rot = np.r_[
```

Comment:
use xy_rotation_matrix() function instead for clarity

Reply:
nice